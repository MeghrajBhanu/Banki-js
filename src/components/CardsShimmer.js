import classes from "./CardsShimmer.module.css"

const CardsShimmer=()=>{
    return(
    <>
        <div className={classes["sort_bar"]}></div>
        <div className={classes["title"]}></div>
        <div className={classes["card"]}>
            {Array(10).fill(0).map((i,j)=>{
            return <div className={classes["shimmer_card"]} key={j}>loading..</div>
        })}
        </div>
    </>
    )
}
export default CardsShimmer;