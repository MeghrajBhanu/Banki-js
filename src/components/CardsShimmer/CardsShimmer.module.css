.shimmer_card{
    
    height:200px;
    width:300px;
    background-color: rgb(143, 142, 138);
    margin:10px;
}
.card{
    margin-top:10px;
    display: flex;
    flex-wrap:wrap;
}
.sort_bar{
    display:block;
    height:45px;
    background-color: rgb(182, 180, 179);
    margin-bottom:25px;
}
.title{
    margin:auto;
    margin-top:64px;
    width:100%;
    height:250px;
    background-color: rgb(143, 142, 138);
    padding:20px;
}
.cards {
    display: flex;
    align-items: flex-start;
    flex-wrap: wrap;
    justify-content: space-around;
  }
  
  .card {
    background-color: rgb(143, 142, 138);
    width: 20em;
    border-radius: 0.8em;
    margin: 1em;
    height:400px;
   
  }