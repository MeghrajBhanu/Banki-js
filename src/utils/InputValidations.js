export const validateEmail = (checkingText) => {
    
    const regexp = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/; 
   
    
    if (regexp.exec(checkingText) !== null) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Email Address'
            };
        }
}
export const validatePan = (checkingText) => {
    
    
    if (checkingText.length === 10) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Pan Number'
            };
        }
}
export const validatePass = (checkingText) => {
    
    
    if (checkingText!=="") {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Password'
            };
        }
}