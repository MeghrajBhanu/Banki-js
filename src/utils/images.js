import bank1 from "../assets/bank1.jpg";
import bank2 from "../assets/bank2.png";
import bank3 from "../assets/bank3.jfif";
import bank4 from "../assets/bank4.png";

const image_map={
    "Bank1":bank1,
    "Bank2":bank2,
    "Bank3":bank3,
    "Bank4":bank4,
}
export default image_map;