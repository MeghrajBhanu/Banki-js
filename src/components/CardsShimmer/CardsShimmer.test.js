import { render, screen, waitFor } from "@testing-library/react";
import { onlywrap, renderComponent } from "../../utils/test_helper";
import renderer from "react-test-renderer";
import CardsShimmer from "./CardsShimmer";

describe("CardsShimmer", () => {
  test("snapshot test", () => {
    const tree = renderer.create(onlywrap(<CardsShimmer />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});