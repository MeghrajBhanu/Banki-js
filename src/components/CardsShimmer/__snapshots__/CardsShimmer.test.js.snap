// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsShimmer snapshot test 1`] = `
Array [
  <div
    className="title"
  />,
  <div
    className="sort_bar"
  />,
  <div
    className="cards"
  >
    <div
      className="card"
    >
      loading..
    </div>
    <div
      className="card"
    >
      loading..
    </div>
    <div
      className="card"
    >
      loading..
    </div>
    <div
      className="card"
    >
      loading..
    </div>
    <div
      className="card"
    >
      loading..
    </div>
    <div
      className="card"
    >
      loading..
    </div>
    <div
      className="card"
    >
      loading..
    </div>
    <div
      className="card"
    >
      loading..
    </div>
    <div
      className="card"
    >
      loading..
    </div>
    <div
      className="card"
    >
      loading..
    </div>
  </div>,
]
`;
